tools/build_defs/repo/http.bzl

```diff
@@ -38,103 +38,35 @@ replace the native rules.
     "patch",
     "read_netrc",
     "read_user_netrc",
     "update_attrs",
     "use_netrc",
     "workspace_and_buildfile",
 )
-
-# Shared between http_jar, http_file and http_archive.
-
-_URL_DOC = """A URL to a file that will be made available to Bazel.
-
-This must be a file, http or https URL. Redirections are followed.
-Authentication is not supported.
-
-More flexibility can be achieved by the urls parameter that allows
-to specify alternative URLs to fetch from."""
-
-_URLS_DOC = """A list of URLs to a file that will be made available to Bazel.
-
-Each entry must be a file, http or https URL. Redirections are followed.
-Authentication is not supported.
-
-URLs are tried in order until one succeeds, so you should list local mirrors first.
-If all downloads fail, the rule will fail."""
-
-def _get_all_urls(ctx):
-    """Returns all urls provided via the url or urls attributes.
-
-    Also checks that at least one url is provided."""
-    if not ctx.attr.url and not ctx.attr.urls:
-        fail("At least one of url and urls must be provided")
-
-    all_urls = []
-    if ctx.attr.urls:
-        all_urls = ctx.attr.urls
-    if ctx.attr.url:
-        all_urls = [ctx.attr.url] + all_urls
-
-    return all_urls
-
-_AUTH_PATTERN_DOC = """An optional dict mapping host names to custom authorization patterns.
-
-If a URL's host name is present in this dict the value will be used as a pattern when
-generating the authorization header for the http request. This enables the use of custom
-authorization schemes used in a lot of common cloud storage providers.
-
-The pattern currently supports 2 tokens: <code>&lt;login&gt;</code> and
-<code>&lt;password&gt;</code>, which are replaced with their equivalent value
-in the netrc file for the same host name. After formatting, the result is set
-as the value for the <code>Authorization</code> field of the HTTP request.
-
-Example attribute and netrc for a http download to an oauth2 enabled API using a bearer token:
-
-<pre>
-auth_patterns = {
-    "storage.cloudprovider.com": "Bearer &lt;password&gt;"
-}
-</pre>
-
-netrc:
-
-<pre>
-machine storage.cloudprovider.com
-        password RANDOM-TOKEN
-</pre>
-
-The final HTTP request would have the following header:
-
-<pre>
-Authorization: Bearer RANDOM-TOKEN
-</pre>
-"""
-
-def _get_auth(ctx, urls):
-    """Given the list of URLs obtain the correct auth dict."""
-    if ctx.attr.netrc:
-        netrc = read_netrc(ctx, ctx.attr.netrc)
-    elif "NETRC" in ctx.os.environ:
-        netrc = read_netrc(ctx, ctx.os.environ["NETRC"])
-    else:
-        netrc = read_user_netrc(ctx)
-    return use_netrc(netrc, urls, ctx.attr.auth_patterns)
-
-def _update_sha256_attr(ctx, attrs, download_info):
-    # We don't need to override the sha256 attribute if integrity is already specified.
-    sha256_override = {} if ctx.attr.integrity else {"sha256": download_info.sha256}
-    return update_attrs(ctx.attr, attrs.keys(), sha256_override)
+load(
```

Comment:
I'm a tiny bit nervous about this. If anything in http_common.bzl loads from any repo, we'd run into trouble again. AFAICT we only need the rules_java label manipulation stuff in a separate package. Could we only extract that part out, and keep the rest of `_http_jar_impl` in this package (at least for now)?